application/src/Service/RdfImporter.php

```diff
@@ -163,25 +118,156 @@ protected function extractMembers(
+     * @param array $vocab Vocab info supported by the vocabulary entity adapter
+     * @param array $options
+     * @return Omeka\Api\Response
+     */
+    public function import($strategy, array $vocab, array $options = [])
+    {
+        $members = $this->getMembers($strategy, $vocab['o:namespace_uri'], $options);
+
+        // Convert to format that the API understands.
+        $vocabMembers = ['o:class' => [], 'o:property' => []];
+        foreach ($members['classes'] as $localName => $info) {
+            $vocabMembers['o:class'][] = [
+                'o:local_name' => $localName,
+                'o:label' => $info['label'],
+                'o:comment' => $info['comment'],
+            ];
+        }
+        foreach ($members['properties'] as $localName => $info) {
+            $vocabMembers['o:property'][] = [
+                'o:local_name' => $localName,
+                'o:label' => $info['label'],
+                'o:comment' => $info['comment'],
+            ];
+        }
+
+        return $this->apiManager->create('vocabularies', array_merge($vocab, $vocabMembers));
+    }
+
+    /**
+     * Get the diff between a stored vocab and one represented in an RDF graph.
+     *
+     * Only gets the diffs that we can process safely: add new members and
+     * update existing labels and comments. It doesn't get the diffs we cannot
+     * process safely: delete existing members or update existing local names.
+     *
+     * @see self::getMembers()
+     * @param string $strategy
+     * @param string $namespaceUri
+     * @param array $options
+     * @return array
+     */
+    public function getDiff($strategy, $namespaceUri, array $options = [])
+    {
+        // Get classes and properties from the database.
+        $classes = $this->apiManager->search(
+            'resource_classes',
+            ['vocabulary_namespace_uri' => $namespaceUri]
+        )->getContent();
+        $dbClasses = [];
+        foreach ($classes as $class) {
+            $dbClasses[$class->localName()] = [
+                'label' => $class->label(),
+                'comment' => $class->comment(),
+            ];
+        }
+        $properties = $this->apiManager->search(
+            'properties',
+            ['vocabulary_namespace_uri' => $namespaceUri]
+        )->getContent();
+        $dbProperties = [];
+        foreach ($properties as $property) {
+            $dbProperties[$property->localName()] = [
+                'label' => $property->label(),
+                'comment' => $property->comment(),
+            ];
+        }
+
+        // Get classes and properties from the RDF graph.
+        $members = $this->getMembers($strategy, $namespaceUri, $options);
+
+        // Extract differences.
+        $diff = [
+            'classes' => ['new' => [], 'label' => [], 'comment' => []],
+            'properties' => ['new' => [], 'label' => [], 'comment' => []],
+        ];
+        foreach ($members['classes'] as $localName => $info) {
```

Comment:
I believe these two foreaches are exactly the same, just using different input and output arrays... it's probably worth it to consolidate them to a function that takes those as params.